cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSICKANSink.java

```diff
@@ -442,22 +450,27 @@ public String buildPkgName(String fiwareService, String fiwareServicePath) throw
      * @return Resource name
      * @throws CygnusBadConfiguration
      */
-    public String buildResName(String entity) throws CygnusBadConfiguration {
+    public String buildResName(String entity, String entityId) throws CygnusBadConfiguration {
         String resName;
-        
-        if (enableEncoding) {
-            resName = NGSICharsets.encodeCKAN(entity);
-        } else {
-            resName = NGSIUtils.encode(entity, false, true).toLowerCase(Locale.ENGLISH);
-        } // if else
+        switch(dataModel) {
+            case DMBYENTITYID:
+            	resName=entityId;
+                break;
+        default:
+            if (enableEncoding) {
+                resName = NGSICharsets.encodeCKAN(entity);
+            } else {
+                resName = NGSIUtils.encode(entity, false, true).toLowerCase(Locale.ENGLISH);
+            } // if else
```

Comment:
Block seems to be missindented

![imagen](https://user-images.githubusercontent.com/1534240/78289580-ae30e800-7522-11ea-9ef1-9f7f55473679.png)

Mabye some spaces vs. tab issue?

Reply:
Fixed in [1651f0d](https://github.com/telefonicaid/fiware-cygnus/pull/1850/commits/1651f0dbeb116b8222d7fe9df90bd4d603b17b5e)